src/pyDMON.py

```diff
@@ -2760,45 +2698,22 @@ def post(self, hostFQDN):
 
         os.environ['ES_HEAP_SIZE'] = qESCoreStart.ESCoreHeap
 
-        # check for running detached es core
-        if os.path.isfile(pidESLoc):
-            esPIDf = check_proc(pidESLoc)
-        else:
-            esPIDf = 0
-
-        if esPIDf != qESCoreStart.ESCorePID:
-            app.logger.warning("[%s] : [WARN] Conflicting PID values found, detached pid -> %s, attached -> %s",
-                               datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d %H:%M:%S'), str(esPIDf),
-                               str(qESCoreStart.ESCorePID))
-        elif checkPID(int(esPIDf)) is True:
-            app.logger.info("[%s] : [INFO] ES Core alredy running with detached pid %s",
-                            datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d %H:%M:%S'),
-                            str(qESCoreStart.ESCorePID))
-            response = jsonify({'Status': 'Detected detached ES Core instance', 'PID': esPIDf})
-            response.status_code = 200
-            return response
-        else:
-            try:
-                subprocess.check_call(["service", "dmon-es", "start", qESCoreStart.ESCoreHeap])
-            except Exception as inst:
-                app.logger.error("[%s] : [ERROR] Cannot start ES Core service with %s and %s",
-                                 datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d %H:%M:%S'), type(inst),
-                                 inst.args)
-                response = jsonify({'Status': 'Error', 'Message': 'Cannot start ES Core'})
-                response.status_code = 500
-                return response
-            esPID = check_proc(pidESLoc)
-            if not esPID:
-                app.logger.error("[%s] : [ERROR] Can't read pidfile for es core",
-                                 datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d %H:%M:%S'))
-                response = jsonify({'Status': 'Error', 'Message': 'Cannot read escore pid file'})
-                response.status_code = 500
-                return response
-            qESCoreStart.ESCorePID = esPID
-            qESCoreStart.ESCoreStatus = 'Running'
-            response = jsonify({'Status': 'ES Core Started', 'PID': esPID})
-            response.status_code = 201
+        try:
+            subprocess.check_call(["service", "dmon-es", "restart",
```

Comment:
Maybe it would be safer to call "initctl restart dmon-es ..." directly here in order to avoid service wrapper? I'm a bit worried that service wrapper might not pass all of the parameters to the upstart and things could get messy and hard to debug.